Assignment4/CountingIslands.java

```diff
@@ -0,0 +1,58 @@
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+ 
+public class CountingIslands
+{
+    int CountingIslands(int noRows, int noCols, boolean[][] landOrWater) 
+	{
+		int index, indexj;
+		// Make a bool array to mark visited cells.
+	    boolean visited[][] = new boolean[noRows][noCols];
+	 
+	    // Count the islands
+	    int count = 0;
+
+	    for (index = 0; index < noRows; index++)
```

Comment:
You could use 'row' or 'r' or something that makes the variable tie back to it being a row-counter. "index" is not very helpful. You could just use "i", as it's about as informative.

Short names are fine if they're obvious. Long, unhelpful names are never good. :)